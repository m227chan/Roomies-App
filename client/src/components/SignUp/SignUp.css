.root {
  background-color: #000000;
  opacity: 0.9;
  overflow: hidden;
}

.mainMessage {
  opacity: 0.9;
}

.mainContainer {
  margin: 0;
  background-color: var(--color-bg-variant);
}

.title {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1.5rem 0rem;
  color: var(--color-primary);
  background-color: var(--color-bg-variant);
}

.newButton {
  background-color: var(--color-bg-variant);
}

.divider {
  background: var(--color-primary);
}

.orDivider {
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: bold;
  width: 100%;
  padding: 0.5rem 0.2rem;
  color: #02473B;
}

.orSectionGrid {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: .5rem;
}

.messageBox {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 2rem 0rem;
  width: auto;
  font-weight: bold;
  color: #02473B;
}


@media (max-width: 600px) {
  .mainMessageContainer {
    margin-left: 4px;
  }
}

.paper {
  overflow: hidden;
  box-shadow: none;
}