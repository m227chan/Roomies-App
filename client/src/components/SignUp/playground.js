import React, { useState } from "react";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";
import history from "../Navigation/history";

import { withStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { MuiThemeProvider, createTheme } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";

import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Container from "@material-ui/core/Container";

import {
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
} from "firebase/auth";
import { auth } from "../Firebase/firebase";

const serverURL = "https://roomies-app.netlify.app"; //enable for dev mode
// const serverURL ="http://ec2-18-216-101-119.us-east-2.compute.amazonaws.com:3006";

const opacityValue = 0.9;

const theme = createTheme({
  palette: {
    type: "light",
    background: {
      default: "#000000",
    },
    primary: {
      main: "#7a1a06",
    },
    secondary: {
      main: "#ffcb52",
    },
  },
});

const styles = (theme) => ({
  root: {
    body: {
      backgroundColor: "#000000",
      opacity: opacityValue,
      overflow: "hidden",
    },
  },
  mainMessage: {
    opacity: opacityValue,
  },

  mainMessageContainer: {
    marginTop: "15vh",
    marginLeft: theme.spacing(20),
    [theme.breakpoints.down("xs")]: {
      marginLeft: theme.spacing(4),
    },
  },
  paper: {
    overflow: "hidden",
  },
  message: {
    opacity: opacityValue,
    maxWidth: 250,
    paddingBottom: theme.spacing(2),
  },
});

const SignUp = (props) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");

  const [submitClicked, setSubmitClicked] = useState(false);
  const [message, setMessage] = useState("");

  const { classes } = props;

  const onSubmit = async () => {
    // console.log("email: " + email + " | " + "password: " + password);
    setSubmitClicked(true);
    try {
      setMessage("");
      await createUserWithEmailAndPassword(auth, email, password);

      await signInWithEmailAndPassword(auth, email, password);

      setMessage("An account has been created.");
      callApiAddUser();
      history.push("/JoinCreateRoom");
    } catch (e) {
      console.log(e.message);
      setMessage(e.message.replace("Firebase: ", ""));
    }
  };

  const callApiAddUser = async () => {
    const user = auth.currentUser;
    // console.log(user.uid);

    const url = serverURL + "/api/addUser";
    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        //authorization: `Bearer ${this.state.token}`
      },
      body: JSON.stringify({
        firstName: firstName,
        lastName: lastName,
        firebaseUID: user.uid,
      }),
    });
    const body = await response.json();
    if (response.status !== 200) throw Error(body.message);
    // console.log("User settings: ", body);
    return body;
  };

  const mainMessage = (
    <Box sx={{ flexGrow: 1 }}>
      <Grid
        container
        spacing={0}
        direction="column"
        alignItems="flex-start"
        style={{ minHeight: "100vh" }}
        className={classes.mainMessageContainer}
      >
        <Grid item>
          <Container maxWidth="xs">
            <form noValidate onSubmit={onSubmit}>
              <Typography variant="h4" color="primary">
                Create an Account
              </Typography>

              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 50 }}
                required
                fullWidth
                id="email"
                value={email}
                onChange={(event) => {
                  setEmail(event.target.value);
                }}
                label="Email Address"
                name="email"
                autoComplete="email"
                autoFocus
                InputLabelProps={{ shrink: true }}
                error={email === "" && submitClicked === true}
                helperText={
                  email === "" && submitClicked === true
                    ? "Please enter an email."
                    : ""
                }
              />

              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 50 }}
                required
                fullWidth
                name="password"
                value={password}
                onChange={(event) => {
                  setPassword(event.target.value);
                }}
                label="Create Password"
                type="password"
                id="password"
                autoComplete="current-password"
                InputLabelProps={{ shrink: true }}
                error={password === "" && submitClicked === true}
                helperText={
                  password === "" && submitClicked === true
                    ? "Please enter a password."
                    : ""
                }
              />

              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 20 }}
                required
                fullWidth
                name="firstName"
                value={firstName}
                onChange={(event) => {
                  setFirstName(event.target.value);
                }}
                label="First Name"
                type="firstName"
                id="firstName"
                autoComplete="firstName"
                InputLabelProps={{ shrink: true }}
                error={firstName === "" && submitClicked === true}
                helperText={
                  firstName === "" && submitClicked === true
                    ? "Please enter your first name."
                    : ""
                }
              />

              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 20 }}
                required
                fullWidth
                name="lastName"
                value={lastName}
                onChange={(event) => {
                  setLastName(event.target.value);
                }}
                label="Last Name"
                type="lastName"
                id="lastName"
                autoComplete="lastName"
                InputLabelProps={{ shrink: true }}
                error={lastName === "" && submitClicked === true}
                helperText={
                  lastName === "" && submitClicked === true
                    ? "Please enter your last name."
                    : ""
                }
              />

              <Button variant="contained" onClick={onSubmit}>
                Register
              </Button>

              <Typography>{message !== "" ? message : ""}</Typography>
            </form>

            <br />

            <Button color="inherit" variant="contained">
              <Link
                color="inherit"
                style={{ cursor: "pointer" }}
                onClick={() => history.push("/SignIn")}
              >
                <Typography>Sign In</Typography>
              </Link>
            </Button>
          </Container>
        </Grid>
      </Grid>
    </Box>
  );

  return (
    <MuiThemeProvider theme={theme}>
      <div className={classes.root}>
        <CssBaseline />
        <Paper className={classes.paper}>{mainMessage}</Paper>
      </div>
    </MuiThemeProvider>
  );
};

export default withStyles(styles)(SignUp);
