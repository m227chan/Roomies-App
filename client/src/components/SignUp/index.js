import * as React from "react";
import { useState } from "react";

import { auth } from "../Firebase/firebase";

import Button from "@mui/material/Button";
import CssBaseline from "@mui/material/CssBaseline";
import TextField from "@mui/material/TextField";
import Paper from "@material-ui/core/Paper";
import Grid from "@mui/material/Grid";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";

import Typography from "@mui/material/Typography";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import Divider from "@material-ui/core/Divider";
import "./SignUp.css";

import {
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  updateProfile,
} from "firebase/auth";

const serverURL = "http://localhost:3000/"; //enable for dev mode

const theme = createTheme({
  palette: {
    primary: {
      main: "#02473B",
    },
  },
  typography: {
    button: {
      textTransform: "none",
    },
  },
});

const SignUp = ({ history }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");

  const [submitClicked, setSubmitClicked] = useState(false);
  const [message, setMessage] = useState("");

  const onSubmit = async () => {
    // console.log("email: " + email + " | " + "password: " + password);
    setSubmitClicked(true);
    try {
      setMessage("");
      await createUserWithEmailAndPassword(auth, email, password);

      const name = firstName + " " + lastName;

      await updateProfile(auth.currentUser, {
        displayName: name,
      });

      await signInWithEmailAndPassword(auth, email, password);

      setMessage("An account has been created.");
      callApiAddUser();
      history.push("/JoinCreateRoom");
    } catch (e) {
      console.log(e.message);
      setMessage(e.message.replace("Firebase: ", ""));
    }
  };

  const callApiAddUser = async () => {
    const user = auth.currentUser;
    // console.log(user.uid);

    const url = serverURL + "/api/addUser";
    const response = await fetch(url, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        //authorization: `Bearer ${this.state.token}`
      },
      body: JSON.stringify({
        firstName: firstName,
        lastName: lastName,
        firebaseUID: user.uid,
      }),
    });
    const body = await response.json();
    if (response.status !== 200) throw Error(body.message);
    // console.log("User settings: ", body);
    return body;
  };

  const mainMessage = (
    <>
      <Grid className="mainContainer">
        <Grid class="title" item>
          <Box
            sx={{
              height: "20%",
              width: "20%",
            }}
            component="img"
            alt="Roomies App Logo"
            src="/Logo.png"
          />
        </Grid>
        <Grid class="divider" item>
          {/* <Divider /> */}
        </Grid>
      </Grid>
      <Container component="main" maxWidth="sm">
        <Grid class="messageBox" item>
          <Typography variant="p">Hurry your roommates are waiting!</Typography>
        </Grid>
        <Box
          sx={{
            mt: "0",
            pt: "0",
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            justifyContent: "space-between",
          }}
        >
          <Box component="form" noValidate>
            <Button
              variant="contained"
              onClick={() => history.push("/SignIn")}
              fullWidth
              sx={{
                mb: 2,
                bgcolor: "#68B984",
                borderRadius: "50px",
                padding: "0.8rem",
                "&:hover": {
                  cursor: "pointer",
                  bgcolor: "#448E5E",
                  // color: "#68B984",
                },
              }}
            >
              <Typography variant="p" sx={{ fontWeight: "bold" }}>
                {" "}
                Sign in to your Account
              </Typography>
            </Button>
            <Grid container class="orSectionGrid">
              <Grid item xs={4}>
                <Divider style={{ background: "#000000" }} />
              </Grid>
              <Grid item xs={5}>
                <Typography variant="h5" class="orDivider">
                  Create your Account
                </Typography>
              </Grid>
              <Grid item xs={4}>
                <Divider style={{ background: "#000000" }} />
              </Grid>
            </Grid>

            <form noValidate onSubmit={onSubmit}>
              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 20 }}
                required
                fullWidth
                name="firstName"
                value={firstName}
                onChange={(event) => {
                  setFirstName(event.target.value);
                }}
                label="First Name"
                type="firstName"
                id="firstName"
                autoComplete="firstName"
                InputLabelProps={{ shrink: true }}
                error={firstName === "" && submitClicked === true}
                helperText={
                  firstName === "" && submitClicked === true
                    ? "Please enter your first name."
                    : ""
                }
              />

              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 20 }}
                required
                fullWidth
                name="lastName"
                value={lastName}
                onChange={(event) => {
                  setLastName(event.target.value);
                }}
                label="Last Name"
                type="lastName"
                id="lastName"
                autoComplete="lastName"
                InputLabelProps={{ shrink: true }}
                error={lastName === "" && submitClicked === true}
                helperText={
                  lastName === "" && submitClicked === true
                    ? "Please enter your last name."
                    : ""
                }
              />
              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 50 }}
                required
                fullWidth
                id="email"
                value={email}
                onChange={(event) => {
                  setEmail(event.target.value);
                }}
                label="Email Address"
                name="email"
                autoComplete="email"
                autoFocus
                InputLabelProps={{ shrink: true }}
                error={email === "" && submitClicked === true}
                helperText={
                  email === "" && submitClicked === true
                    ? "Please enter an email."
                    : ""
                }
              />
              <TextField
                variant="outlined"
                margin="normal"
                inputProps={{ maxLength: 50 }}
                required
                fullWidth
                name="password"
                value={password}
                onChange={(event) => {
                  setPassword(event.target.value);
                }}
                label="Password"
                type="password"
                id="password"
                autoComplete="current-password"
                InputLabelProps={{ shrink: true }}
                error={password === "" && submitClicked === true}
                helperText={
                  password === "" && submitClicked === true
                    ? "Please enter a password."
                    : ""
                }
              />

              <Grid
                container
                direction="row"
                justifyContent="flex-end"
                alignItems="center"
              >
                {/* <Button
                  variant="contained"
                  onClick={onSubmit}
                  sx={{
                    mt: 3,
                    mb: 2,
                    bgcolor: "#02473B",
                    borderRadius: "50px",
                    paddingLeft: "10%",
                    paddingRight: "10%",
                    "&:hover": {
                      cursor: "pointer",
                      bgcolor: "#448E5E",
                      //   color: "#68B984",
                    },
                  }}
                >
                  <Typography>Register</Typography>
                </Button> */}

                <Typography>{message !== "" ? message : ""}</Typography>

                <Button
                  variant="contained"
                  onClick={onSubmit}
                  sx={{
                    mt: 3,
                    mb: 2,
                    bgcolor: "#02473B",
                    borderRadius: "50px",
                    paddingLeft: "10%",
                    paddingRight: "10%",
                    "&:hover": {
                      cursor: "pointer",
                      bgcolor: "#448E5E",
                      //   color: "#68B984",
                    },
                  }}
                >
                  <Typography>Register</Typography>
                </Button>
              </Grid>
            </form>
          </Box>
        </Box>
      </Container>
    </>
  );

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Paper class="paper">{mainMessage}</Paper>
    </ThemeProvider>
  );
};

export default SignUp;
