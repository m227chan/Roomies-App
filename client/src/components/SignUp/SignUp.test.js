import React from 'react';
import renderer from 'react-test-renderer';
import SignUp from './index';

describe('SignUp', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<SignUp />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});