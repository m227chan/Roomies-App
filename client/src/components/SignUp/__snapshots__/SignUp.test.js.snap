// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<div
  className="SignUp-root-1"
>
  <div
    className="MuiPaper-root SignUp-paper-4 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiBox-root MuiBox-root-6"
    >
      <div
        className="MuiGrid-root SignUp-mainMessageContainer-3 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
        style={
          Object {
            "minHeight": "100vh",
          }
        }
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiContainer-root MuiContainer-maxWidthXs"
          >
            <form
              noValidate={true}
              onSubmit={[Function]}
            >
              <h4
                className="MuiTypography-root MuiTypography-h4 MuiTypography-colorPrimary"
              >
                Create an Account
              </h4>
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={true}
                  htmlFor="email"
                  id="email-label"
                >
                  Email Address
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="email"
                    autoFocus={true}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="email"
                    maxLength={50}
                    name="email"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                    >
                      <span>
                        Email Address
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={true}
                  htmlFor="password"
                  id="password-label"
                >
                  Create Password
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="current-password"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="password"
                    maxLength={50}
                    name="password"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                    >
                      <span>
                        Create Password
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={true}
                  htmlFor="firstName"
                  id="firstName-label"
                >
                  First Name
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="firstName"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="firstName"
                    maxLength={20}
                    name="firstName"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="firstName"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                    >
                      <span>
                        First Name
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                
              </div>
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={true}
                  htmlFor="lastName"
                  id="lastName-label"
                >
                  Last Name
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="lastName"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="lastName"
                    maxLength={20}
                    name="lastName"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="lastName"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="PrivateNotchedOutline-legendLabelled-9 PrivateNotchedOutline-legendNotched-10"
                    >
                      <span>
                        Last Name
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
                
              </div>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Register
                </span>
              </button>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                
              </p>
            </form>
            <br />
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-colorInherit"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <a
                  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <p
                    className="MuiTypography-root MuiTypography-body1"
                  >
                    Sign In
                  </p>
                </a>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
